cli/cmd/resources/odiglet.go

```diff
@@ -505,6 +510,84 @@ func NewOdigletDaemonSet(ns string, version string, imagePrefix string, imageNam
 			},
 		},
 	}
+
+	// Check for a custom offsets file, if present use it
+	// The reason for this is because a user can update their offsets asynchronously with the Odigos release
+	// For example:
+	// 1) User updates offsets to support Go 1.24
+	// 2) Odigos releases support for Go 1.24
+	// 3) User updates offsets to support Go 1.24.1
+	// 4) User updates Odigos (no default support for 1.24.1)
+	// This also means a user could be still using old offsets, and might want to revert that to pull in latest Odigos changes
+	// TODO: Detect superceded offsets automatically so user doesn't need to manually revert
+	_, err := client.Clientset.CoreV1().ConfigMaps(ns).Get(ctx, consts.GoOffsetsConfigMap, metav1.GetOptions{})
+	if err == nil {
+		// Add offsets volume (if not already exist)
+		volumes := ds.Spec.Template.Spec.Volumes
+		if volumes == nil {
+			volumes = make([]v1.Volume, 0)
+		}
+		addVolume := true
+		for _, vol := range volumes {
+			if vol.Name == consts.GoOffsetsConfigMap {
```

Comment:
can this really happen? the volumes is the array that we created right above which will never contain this config map.

consider instead of mutating the ds right after creating it, to have `extraVolume` array with 0 or 1 elements which you create at the start of this function, and then append it with `...` spread right into the volumes in the manifest. and also for the rest of the changes here.

I think it will be more maintainable this way and will align better with the other existing implementations in this package